src/NUnitEngine/nunit.engine/Internal/TestPackageExtensions.cs

```diff
@@ -43,6 +43,13 @@ public static bool HasSubPackages(this TestPackage package)
             return package.SubPackages.Count > 0;
         }
 
+        public static bool IsProjectPackage(this TestPackage package)
+        {
+            // We may not have the project service available, so we assume that any expanded package
+            // is a project package. Check name to verify that it's not the anonymous command-line package.
+            return package.FullName != null && package.HasSubPackages();
```

Comment:
Could just be `return package?.HasSubPackages()`

Reply:
This is not the same. The code examines if `FullName` is null and will throw a NRE if `package` is null. Also `package?.HasSubPackages()` returns a nullable boolean and not a bool, so if the original code had been `package != null && package.HasSubPackages()` then we could have translated that into `package.?HasSubPackages() ?? false`.

Reply:
A package with a null name is essentially the command-line package, which contains all the other packages. So it's not a project.